src/azure-cli/azure/cli/command_modules/aro/custom.py

```diff
@@ -0,0 +1,150 @@
+
+    rp_client_id = FP_CLIENT_ID
+
+    rp_client_sp = aad.get_service_principal(rp_client_id)
+
+    assign_contributor_to_vnet(cmd.cli_ctx, vnet, client_sp.object_id, 'client', tags)
+    assign_contributor_to_vnet(cmd.cli_ctx, vnet, rp_client_sp.object_id, 'rp_client', tags)
+
+    oc = v2020_04_30.OpenShiftCluster(
+        location=location,
+        tags=tags,
+        cluster_profile=v2020_04_30.ClusterProfile(
+            pull_secret=pull_secret or "",
+            domain=domain or random_id,
+            resource_group_id='/subscriptions/%s/resourceGroups/%s' %
+            (subscription_id, cluster_resource_group or "aro-" + random_id),
+        ),
+        service_principal_profile=v2020_04_30.ServicePrincipalProfile(
+            client_id=client_id,
+            client_secret=client_secret,
+        ),
+        network_profile=v2020_04_30.NetworkProfile(
+            pod_cidr=pod_cidr or '10.128.0.0/14',
+            service_cidr=service_cidr or '172.30.0.0/16',
+        ),
+        master_profile=v2020_04_30.MasterProfile(
+            vm_size=master_vm_size or 'Standard_D8s_v3',
+            subnet_id=master_subnet,
+        ),
+        worker_profiles=[
+            v2020_04_30.WorkerProfile(
+                name='worker',  # TODO: 'worker' should not be hard-coded
+                vm_size=worker_vm_size or 'Standard_D2s_v3',
+                disk_size_gb=worker_vm_disk_size_gb or 128,
+                subnet_id=worker_subnet,
+                count=worker_count or 3,
+            )
+        ],
+        apiserver_profile=v2020_04_30.APIServerProfile(
+            visibility=apiserver_visibility or 'Public',
+        ),
+        ingress_profiles=[
+            v2020_04_30.IngressProfile(
+                name='default',  # TODO: 'default' should not be hard-coded
+                visibility=ingress_visibility or 'Public',
+            )
+        ],
+    )
+
+    return sdk_no_wait(no_wait, client.create_or_update,
+                       resource_group_name=resource_group_name,
+                       resource_name=resource_name,
+                       parameters=oc)
+
+
+def aro_delete(client, resource_group_name, resource_name, no_wait=False):
+    # TODO: clean up rbac
+
+    return sdk_no_wait(no_wait, client.delete,
+                       resource_group_name=resource_group_name,
+                       resource_name=resource_name)
+
+
+def aro_list(client, resource_group_name=None):
+    if resource_group_name:
+        return client.list_by_resource_group(resource_group_name)
+    return client.list()
+
+
+def aro_show(client, resource_group_name, resource_name):
+    return client.get(resource_group_name, resource_name)
+
+
+def aro_list_credentials(client, resource_group_name, resource_name):
+    return client.list_credentials(resource_group_name, resource_name)
+
+
+def aro_update(client, resource_group_name, resource_name, location=None, no_wait=False):
+    oc = v2020_04_30.OpenShiftCluster(
```

Comment:
this is OpenShiftClusterUpdate upstream - why does it change here?  Using OpenShiftCluster here is almost certainly wrong